### PR TITLE
[10.0] [MIG] l10n_de_steuernummer

### PR DESCRIPTION
**Pull request by pedrobaeza**
Following #17, this is the migration to 10.0 of the module still in PR. The discussion ended that this can be useful and should be merged, but it has not done it yet. We can move forward to this version now, as the customer is moving to this version.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
cc @Tecnativa

**Comment by pedrobaeza** [comment]
@chienandalu @cubells please review

File: l10n_de_steuernummer/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.model
+    def check_vat_de(self, vat):
+        res = False
+        if getattr(vatnumber, 'check_vat_de', None):
+            res = vatnumber.check_vat_de(vat)
+        if not res:
+            # This vat is not a Steue-IdNr., so if it is 10 or 11 digit
```
Comment (cubells) [inline l10n_de_steuernummer/models/res_partner.py]: > This vat is not **an** Steue-IdNr.

File: l10n_de_steuernummer/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            # length then we can assume that this is a old Steuernummer
```
Comment (cubells) [inline l10n_de_steuernummer/models/res_partner.py]: > that this is **an** old Steuernummer

File: l10n_de_steuernummer/README.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+validation of german SteuerNummer following this schema:
+https://de.wikipedia.org/wiki/Steuernummer
+
+
+Usage
+=====
+
+#. Go to Contacts.
+#. Create a new record.
+#. Put a SteuerNummer in the VAT field preceded by 'DE'.
```
Comment (cubells) [inline l10n_de_steuernummer/README.rst:19]: > Put **an** SteuerNummer in the VAT

**Review by cubells — CHANGES_REQUESTED** [review]
Minor changes. Typos.

File: l10n_de_steuernummer/models/res_partner.py (hunk 1 above)
Comment (chienandalu) [reply l10n_de_steuernummer/models/res_partner.py]: @cubells I think that's not correct. Following grammar rules(https://rayli.net/blog/life/when-to-use-a-or-an/#First_why_the_confusion) a word like Steuer doesn't begin with a vowel sound (although we spanish speakers tend to pronounce such words more like 'es...', than the proper 'ss').

Comment (chienandalu) [reply l10n_de_steuernummer/models/res_partner.py]: Anyway s/Steue-IdNr/Steuer-IdNr

Comment (cubells) [reply l10n_de_steuernummer/models/res_partner.py]: @chienandalu 
I'm not an expert english speaker, sure.

So according to you, I made **"a stupid comment"** or **"an stupid comment"**?
:)

Comment (chienandalu) [reply l10n_de_steuernummer/models/res_partner.py]: @cubells Just an unfortunate observation :smile:

**Review by chienandalu — COMMENTED** [review]
@pedrobaeza Runbot doesn't build. Rebase maybe?

**Comment by pedrobaeza** [comment]
The problem is an outdated .travis.yml. You can update it in a separate PR.

**Comment by chienandalu** [comment]
@pedrobaeza Here you are https://github.com/OCA/l10n-germany/pull/24

**Comment by pedrobaeza** [comment]
Please rebase and fix the comments that should be fixed (if any pending)

**Comment by pedrobaeza** [comment]
@chienandalu @cubells give then your final blessing for merging.

**Comment by chienandalu** [comment]
@pedrobaeza It still skips building runbot :confused:

**Review by cubells — APPROVED** [review]
:+1: 
LGTM

**Comment by pedrobaeza** [comment]
But it's because you haven't rebased properly. Check the .travis.yml in the branch: https://github.com/Tecnativa/l10n-germany/blob/10.0-l10n_de_steuernummer/.travis.yml

**Review by chienandalu — APPROVED** [review]
@pedrobaeza Yes, I didn't update remotes.

Tested on runbot

**Pull request merged**